src/tests/plugins/sendmail/test_rules.py

```diff
@@ -276,12 +276,120 @@ def test_sendmail_rule_send_correct_products(event, order, item, item2):
     assert djmail.outbox[0].to[0] == p1.attendee_email
 
 
+@pytest.mark.django_db
+@scopes_disabled()
+def run_restriction_test(event, order, restrictions_pass=[], restrictions_fail=[]):
+    for r in restrictions_pass:
+        djmail.outbox = []
+        event.sendmail_rules.create(send_date=dt_now - datetime.timedelta(hours=1), restrict_to_status=[r],
+                                    subject='meow', template='meow meow meow')
+        sendmail_run_rules(None)
+
+        assert len(djmail.outbox) == 1
+
+    for r in restrictions_fail:
+        djmail.outbox = []
+        event.sendmail_rules.create(send_date=dt_now - datetime.timedelta(hours=1), restrict_to_status=[r],
+                                    subject='meow', template='meow meow meow')
+        sendmail_run_rules(None)
+
+        assert len(djmail.outbox) == 0
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_paid(event, order):
+    order.status = Order.STATUS_PAID
+    order.save()
+    order.valid_if_pending = False
+    restrictions_pass = ['p']
+    restrictions_fail = ['e', 'c', 'na', 'pa', 'valid_if_pending', 'overdue']
+
+    run_restriction_test(event, order, restrictions_pass, restrictions_fail)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_pending(event, order):
+    order.status = Order.STATUS_PENDING
+    order.require_approval = False
+    order.valid_if_pending = False
+    order.save()
+    restrictions_pass = ['na']
+    restrictions_fail = ['p', 'e', 'c', 'pa', 'valid_if_pending', 'overdue']
+
+    run_restriction_test(event, order, restrictions_pass, restrictions_fail)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_valid_pending(event, order):
+    order.status = Order.STATUS_PENDING
+    order.require_approval = False
+    order.valid_if_pending = True
+    order.save()
+    restrictions_pass = ['valid_if_pending']
+    restrictions_fail = ['p', 'e', 'c', 'na', 'pa', 'overdue']
+
+    run_restriction_test(event, order, restrictions_pass, restrictions_fail)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_approval_pending(event, order):
+    order.status = Order.STATUS_PENDING
+    order.require_approval = True
+    order.valid_if_pending = False
+    order.save()
+    restrictions_pass = ['pa']
+    restrictions_fail = ['p', 'e', 'c', 'na', 'valid_if_pending', 'overdue']
+
+    run_restriction_test(event, order, restrictions_pass, restrictions_fail)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_overdue_pending(event, order):
+    order.status = Order.STATUS_PENDING
+    order.require_approval = True
+    order.valid_if_pending = False
+    order.expires = order.datetime - datetime.timedelta(hours=1)  # TODO: make this test-order actually overdue
```

Comment:
This still needs more thoughts, as either the overdue check is wrong or the actual overdue restriction

Reply:
I think overdue might only be a valid choice if

event.settings.payment_term_expire_automatically = True

is set first?